From 7a70a475f5dc624cb8647a417ede1dc72aa2a088 Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@ti.com>
Date: Thu, 9 Aug 2012 22:02:33 +0800
Subject: [PATCH 052/609] ARM:OMAP:AM33XX: Add PROFIBUS support

Original commit: c0244b80275a7f5c42603aef6ba87c7218db45da

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

  Added PROFIBUS pinmux settings

Signed-off-by: Amit Shah <amit.shah@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c |   14 ++++++++++++++
 arch/arm/mach-omap2/mux33xx.c         |    6 +++---
 2 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index d19e271..39b5c8f 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -712,6 +712,14 @@ static struct pinmux_config usb1_pin_mux[] = {
 	{NULL, 0},
 };
 
+/* pinmux for profibus */
+static struct pinmux_config profibus_pin_mux[] = {
+	{"uart1_rxd.pr1_uart0_rxd_mux1", OMAP_MUX_MODE5 | AM33XX_PIN_INPUT},
+	{"uart1_txd.pr1_uart0_txd_mux1", OMAP_MUX_MODE5 | AM33XX_PIN_OUTPUT},
+	{"mcasp0_fsr.pr1_pru0_pru_r30_5", OMAP_MUX_MODE5 | AM33XX_PIN_OUTPUT},
+	{NULL, 0},
+};
+
 /* Module pin mux for eCAP0 */
 static struct pinmux_config ecap0_pin_mux[] = {
 	{"ecap0_in_pwm0_out.gpio0_7", AM33XX_PIN_OUTPUT},
@@ -1217,6 +1225,7 @@ static struct evm_dev_cfg ind_auto_mtrl_evm_dev_cfg[] = {
 	{mii1_init,	DEV_ON_DGHTR_BRD, PROFILE_ALL},
 	{usb0_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{usb1_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
+	{profibus_init, DEV_ON_DGHTR_BRD, PROFILE_ALL},
 	{evm_nand_init, DEV_ON_DGHTR_BRD, PROFILE_ALL},
 	{spi1_init,	DEV_ON_DGHTR_BRD, PROFILE_ALL},
 	{uart3_init,	DEV_ON_DGHTR_BRD, PROFILE_ALL},
@@ -1339,6 +1348,11 @@ static void setup_beaglebone(void)
 	_configure_device(LOW_COST_EVM, beaglebone_dev_cfg, PROFILE_NONE);
 }
 
+static void profibus_init(int evm_id, int profile)
+{
+	setup_pin_mux(profibus_pin_mux);
+	return;
+}
 
 static void am335x_setup_daughter_board(struct memory_accessor *m, void *c)
 {
diff --git a/arch/arm/mach-omap2/mux33xx.c b/arch/arm/mach-omap2/mux33xx.c
index 08d9ae6..e2ba2b4 100644
--- a/arch/arm/mach-omap2/mux33xx.c
+++ b/arch/arm/mach-omap2/mux33xx.c
@@ -322,10 +322,10 @@ static struct omap_mux __initdata am33xx_muxmodes[] = {
 		"spi1_cs1", NULL, NULL, NULL),
 	_AM33XX_MUXENTRY(UART1_RXD, 0,
 		"uart1_rxd", "mmc1_sdwp", NULL, NULL,
-		NULL, NULL, NULL, NULL),
+		NULL, "pr1_uart0_rxd_mux1", NULL, NULL),
 	_AM33XX_MUXENTRY(UART1_TXD, 0,
 		"uart1_txd", "mmc2_sdwp", NULL, NULL,
-		NULL, NULL, NULL, NULL),
+		NULL, "pr1_uart0_txd_mux1", NULL, NULL),
 	_AM33XX_MUXENTRY(I2C0_SDA, 0,
 		"i2c0_sda", NULL, NULL, NULL,
 		NULL, NULL, NULL, NULL),
@@ -349,7 +349,7 @@ static struct omap_mux __initdata am33xx_muxmodes[] = {
 		"mmc0_sdwp", NULL, NULL, NULL),
 	_AM33XX_MUXENTRY(MCASP0_FSR, 0,
 		"mcasp0_fsr", NULL, "mcasp0_axr3", "mcasp1_fsx",
-		NULL, NULL, NULL, NULL),
+		NULL, "pr1_pru0_pru_r30_5", NULL, NULL),
 	_AM33XX_MUXENTRY(MCASP0_AXR1, 0,
 		"mcasp0_axr1", NULL, NULL, "mcasp1_axr0",
 		NULL, NULL, NULL, NULL),
-- 
1.7.5.4

